source/utils/ethUtil.ts

```diff
@@ -1,106 +1,66 @@
-import { useController } from 'hooks/index';
 import InputDataDecoder from 'ethereum-input-data-decoder';
-import Web3 from 'web3';
-import { KeyringNetwork } from '@stardust-collective/dag4-keyring'
 import store from 'state/store';
 import IVaultState, { AssetType, IAssetState } from 'state/vault/types';
-import erc20ABI from './erc20.json'
-import { getContractDetails } from './etherscan';
-import { EthNetworkId } from 'scripts/Background/controllers/EthChainController/types';
-import { getInfuraUrl } from 'scripts/Background/controllers/EthChainController/utils';
+import erc20ABI from './erc20.json';
+import { getWalletController } from './controllersUtils';
+import { parseUnits } from 'ethers/lib/utils';
 
 export const getERC20DataDecoder = () => {
     return new InputDataDecoder(erc20ABI as any);
 }
 
 export const estimateGasPrice = async (): Promise<number> => {
-    const controller = useController();
+    const walletController = getWalletController();
+    const ethClient = walletController.account.ethClient;
 
-    const client = controller.wallet.account.ethClient;
-
-    if (!client) {
+    if (!ethClient) {
         return 0;
     }
 
     // Returns average, fast, or fastest
-    const { fast } = await client.estimateGasPrices();
+    const { fast } = await ethClient.estimateGasPrices();
     return fast.amount().toNumber();
 }
 
-const _getWeb3 = (network: EthNetworkId) => {
-    const infuraUrl = getInfuraUrl(network);
-
-    return new Web3(
-        new Web3.providers.HttpProvider(`${infuraUrl}${process.env.INFURA_CREDENTIAL}`)
-    );
-}
-
-const _getAbi = async ({ to }: {to: string}) => {
+export const estimateGasLimitForTransfer = async ({ to, from, amount: value, gas }: {to: string, from: string, amount: string, gas: string}): Promise<number> => {
+    const walletController = getWalletController();
+    const ethClient = walletController.account.ethClient;
 
-    const { activeNetwork, activeWallet }: IVaultState = store.getState().vault;
-    const network = activeNetwork[KeyringNetwork.Ethereum] as EthNetworkId;
-    const knownERC20Asset = activeWallet.assets.find((asset: IAssetState) => asset.contractAddress === to && asset.type === AssetType.ERC20);
+    if (gas) return parseInt(gas);
```

Comment:
```suggestion
    if (gas) {
        return parseInt(gas);
    }
```
 long version preferred.